File: Exavision.Seasense/Client/src/app/materials/settings/setting-switch-value-capability.ts
import { SwitchValue } from "../../models/switch-value";
import { CapabilityType } from "../capabilities/capability-type";
import { DoubleValueType } from "../capabilities/double-value-type";
import { SwitchValueType } from "../capabilities/switch-value-type";
import { SettingCapability } from "./setting-capability";

export interface SettingSwitchValueCapability extends SettingCapability {
  id: string;
  capabilityType: CapabilityType;
  switchValueType: SwitchValueType;
  values: SwitchValue[]; 
  displayName: string;
  type: string;

}

File: Exavision.Seasense.Shared/Settings/SettingSwitchValueCapability.cs
using Exavision.Seasense.Shared.Capabilities;
using Exavision.Seasense.Shared.Models;
using System.Collections.Generic;

namespace Exavision.Seasense.Shared.Settings {
    public class SettingSwitchValueCapability : SettingCapability {
        public List<SwitchValue> Values { get; set; } = new List<SwitchValue>();

        public SwitchValueType SwitchValueType { get; set; }
    }
}
